query-status.hh

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright (C) 2020 ScyllaDB
+ */
+
+/*
+ * This file is part of Scylla.
+ *
+ * Scylla is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Scylla is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Scylla.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+#include <cinttypes>
+
+namespace ser {
+template <typename T>
+class serializer;
+}
+
+namespace query {
+
+class status {
+public:
+    enum value { OK, OVERLOADED };
```

Comment:
Why not `enum class status` instead of `class status`?

Reply:
Can enum class have methods? I need something to allow serialization, so I went with a regular class with a method which translates the value to uint8_t. I always avoided enum classes because they ended up being painful instead of easier to use.

Reply:
I thought we already have an enum serializer. Or maybe I just wanted to write one and didn't end up using it. Anyway, it is trivial to write one.